app/controllers/planning_applications_controller.rb

```diff
@@ -71,14 +71,37 @@ def withdraw
     redirect_to @planning_application
   end
 
+  def cancel_confirmation
+    render :cancel_confirmation
+  end
+
+  def cancel
+    status = authorize_user_can_update_status(
+        planning_application_params[:status])
+    apply_cancellation(status)
+    @planning_application.update!(cancellation_comment: planning_application_params[:cancellation_comment])
+    if @planning_application.withdrawn? || @planning_application.returned?
```

Comment:
Is there a situation where this wouldn't be true? Do we have a test for the case where it wouldn't be true? If not, I'd remove this `if` statement, as we always expect it to be true.